pkg/apis/apps/v1alpha1/servicebindingrequest_types.go

```diff
@@ -81,6 +81,48 @@ type ApplicationSelector struct {
 	LabelSelector               *metav1.LabelSelector `json:"labelSelector,omitempty"`
 	metav1.GroupVersionResource `json:",inline"`
 	ResourceRef                 string `json:"resourceRef,omitempty"`
+
+	// BindingPath refers to the path in the application workload's schema
+	// where the binding workload would be referenced.
+	// +optional
+	BindingPath *BindingPath `json:"bindingPath,omitempty"`
+}
+
+const (
+	// DefaultContainersPath has the logical path logical path
+	// to find containers on supported objects
+	// Used as []string{"spec", "template", "spec", "containers"}
+	DefaultPathToContainers = "spec.template.spec.containers"
+
+	// DefaultPathToVolumes is the logical path to find volumes on supported objects
+	// used as []string{"spec", "template", "spec", "volumes"}
+	DefaultPathToVolumes = "spec.template.spec.volumes"
+)
+
+// BindingPath defines the path to the field where the binding would be
+// embedded in the workload
+type BindingPath struct {
+	// PodSpecPath overrides the default podSpec path
+	// +optional
+	PodSpecPath *PodSpecPath `json:"podSpecPath,omitempty"`
```

Comment:
Why do we have additional nesting level for `Containers` and `Volumes`, i.e. what speaks against having something like this:

```go
type BindingPath struct {

	ContainersPath string `json:"containersPath"`

  	VolumesPath string `json:"volumesPath"`

   SecretPath string `json:"secretPath,omitempty"`
}

I would drop "Custom"  from "CustomSecretPath", because it does not add anything on semantic. Setting any value here means that we are specifying a custom path/.
```

Reply:
I'm fine with dropping `Custom` from names, agree with the rationale.

I'm also fine dropping the nesting, but I believe the nesting is there to accommodate other keys similar to `PodSpecPath` in the future; perhaps @sbose78 can clarify the design?

Reply:
This did go through a bit of evolution at time. The initial idea was to only support a custom podSpec path based on the feedback that our design was a bit too strict on the podSpec path. 

Gradually, things unravelled further, we've realized that individual containers, volumes, and secret paths are valuable too. 

I'm good with your suggestion @pedjak and @isutton .

Reply:
The reason podSpecPath and secretpath was separated with the layer of nesting was that, some types cared about podSpec while some didn't at all.

Reply:
When `spec.bindingPath` is unspecified, internally, the defaults should be the podspec defaults
* `containersPath` defaults to the default podspec containers path
* `volumesPath` defaults to the default podspec volumes path
* `secretPath` defaults to the contatiners API's secretRef path.

Example, one of the use cases:
Our code should gracefully handle workload types which care about: 
* only the custom `secretPath` or 
* only the `containersPath` ( Coligo's usecase) or 
* only the `volumesPath`.

Reply:
Done. dbfcd0dab8fbf8fcb3ce2e4c86745537e2bf3f4b